tests/tests_test.go

```diff
@@ -0,0 +1,267 @@
+				},
+			}
+			_, err = sourceClient.AppsV1().Deployments("ocp-41583-longpvcname").Create(ctx, deployment, metav1.CreateOptions{})
+			if err != nil {
+				log.Println(err)
+			}
+			if err != nil {
+				log.Println(err)
+			}
+			Eventually(func() string {
+				pods, err := sourceClient.CoreV1().Pods("ocp-41583-longpvcname").List(ctx, metav1.ListOptions{})
+				if err != nil {
+					log.Println(err)
+				}
+				for _, p := range pods.Items {
+					return string(p.Status.Phase)
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Running"))
+		})
+
+		AfterEach(func() {
+			for _, ns := range namespaces {
+				err = sourceClient.CoreV1().Namespaces().Delete(ctx, ns, metav1.DeleteOptions{})
+				if err != nil {
+					log.Println(err)
+				}
+				err = hostClient.Delete(context.TODO(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
+				if err != nil {
+					log.Println(err)
+				}
+			}
+		})
+
+		It("Should create a new migration", func() {
+			By("Creating a new migmigration")
+
+			migrationName := "foo"
+			migration := NewMigMigration(migrationName, migPlanName, false, false)
+			Expect(hostClient.Create(ctx, migration)).Should(Succeed())
+
+			fooMigration := &migapi.MigMigration{}
+
+			Eventually(func() string {
+				dvms := &migapi.DirectVolumeMigrationList{}
+				err := hostClient.List(ctx, dvms, client.InNamespace(MigrationNamespace))
+				if err != nil {
+					log.Println(err)
+				}
+				for _, dvm := range dvms.Items {
+					if dvm.GenerateName == migrationName+"-" {
+						if dvm.Status.Itinerary == "VolumeMigration" {
+							dvmps := &migapi.DirectVolumeMigrationProgressList{}
+							err = hostClient.List(ctx, dvmps, client.MatchingLabels{"directvolumemigration": string(dvm.UID)})
+							if err != nil {
+								log.Println(err)
+							}
+							for _, dvmp := range dvmps.Items {
+								if dvmp.Status.TotalProgressPercentage == "100%" {
+									return dvm.Status.Phase
+								}
+							}
+						} else {
+							log.Println("Direct volume migration Failed")
+						}
+					}
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+			Eventually(func() string {
+				hostClient.Get(ctx, client.ObjectKey{Name: migrationName, Namespace: MigrationNamespace}, fooMigration)
+				if fooMigration.Status.Itinerary == "Final" {
+					return fooMigration.Status.Phase
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+
+			// TODO check the destination pvc for data
+			Eventually(func() bool {
+				podList := &v1.PodList{}
+				hostClient.List(ctx, podList, &client.ListOptions{
+					LabelSelector: labels.SelectorFromSet(map[string]string{
+						"app": "longpvc-test",
+					},
+					),
+					Namespace: "ocp-41583-longpvcname",
+				})
+				destinationFile := []string{}
+				for _, p := range podList.Items {
+					podCmd := pods.PodCommand{
+						Pod:     &p,
+						RestCfg: hostCfg,
+						Args:    []string{"sh", "-c", "ls -l data/test"},
```

Comment:
@alaypatel07 @pranavgaikwad should we be checksumming the file to make sure it matches what was transferred from the source cluster?

Reply:
Also, just want to confirm: this file is one that is actually transferred via rsync, and not one that will be re-generated after the YAML is moved around, right?

Reply:
@djwhatle Nice catch! I am now creating file on source cluster's pod with exec to make sure the file won't be regenerated after YAML is migrated.

Reply:
@JaydipGabani @djwhatle Is it possible to populate a volume using a temporary pod as a preparation step of the test, and then attaching that same volume in a dummy Pod? so that the Pod being migrated doesn't have to create its own data. We can verify whether we find the same data in the migrated Pod that we created in the prepare step using checksum.